gpytorch/models/exact_prediction_strategies.py

```diff
@@ -145,12 +147,13 @@ def get_fantasy_strategy(self, inputs, targets, full_inputs, full_targets, full_
 
         # Solve for "b", the lower portion of the *new* \\alpha corresponding to the fantasy points.
         schur_complement = fant_fant_covar - fant_train_covar.matmul(fant_solve)
-        if self.mean_cache.dim() == 1:
-            ftcm = fant_train_covar.matmul(self.mean_cache)
-        elif full_inputs[0].dim() <= full_targets.dim():
-            ftcm = torch.einsum("...ij,...j->...i", [fant_train_covar, self.mean_cache])
-        else:
-            ftcm = torch.einsum("f...ij,...j->f...i", [fant_train_covar, self.mean_cache])
+
+        # we'd like to use a less hacky approach for the following, but einsum can be much faster than
+        # than unsqueezing/squeezing here (esp. in backward passes), unfortunately it currenlty has some
+        # issues with broadcasting: https://github.com/pytorch/pytorch/issues/15671
+        prefix = string.ascii_lowercase[:max(fant_train_covar.dim() - self.mean_cache.dim() - 1, 0)]
```

Comment:
String manipulation in an ML package, huh? :-)

Reply:
yep just until torch fixes this...